screens/ChangePassword.screen.jsx

```diff
@@ -13,62 +13,169 @@ import { FButton } from 'components/Buttons/FButton';
 import { View, StyleSheet } from 'react-native';
 import { FImage } from 'components/Composition/FImage';
 import images from 'constants/images';
-import { FKeyboardWrapper } from '../components/Utils/FKeyboardWrapper';
+import { FKeyboardWrapper } from 'components/Utils/FKeyboardWrapper';
+import errorMessages from 'constants/errorMessages';
+import { FModal } from 'components/Composition/FModal';
+import modalTypes from 'constants/modalTypes';
+import { filterErrorMessages } from 'utils/filterErrorMessages';
+import { useErrorModal } from 'hooks/useErrorModal';
+import { updatePasswordService } from '../services/updatePassword.service';
 
-export const ChangePasswordScreen = () => (
-  <FDefaultLayout
-    hasFlatList={false}
-    withLogo={false}
-  >
-    <FKeyboardWrapper>
-      <>
-        <View style={styles.imageContainer}>
-          <FImage
-            imagePath={images.CHANGE_PASSWORD()}
-            width={sizes.WIDTH_120}
-            height={sizes.HEIGHT_120}
-          />
-        </View>
-        <FInput
-          iconPlacement={placements.LEFT}
-          type={inputTypes.PASSWORD}
-          icon={icons.LOCK_CLOSED_OUTLINE}
-          placeholder={locales.PASS_OLD_PASSWORD}
-          marginBottom={sizes.MARGIN_30}
-        />
-        <FInput
-          iconPlacement={placements.LEFT}
-          type={inputTypes.PASSWORD}
-          icon={icons.LOCK_CLOSED_OUTLINE}
-          placeholder={locales.PASS_NEW_PASSWORD}
-        />
-        <FInput
-          iconPlacement={placements.LEFT}
-          type={inputTypes.PASSWORD}
-          icon={icons.LOCK_CLOSED_OUTLINE}
-          placeholder={locales.REPEAT_NEW_PASSWORD}
+export const ChangePasswordScreen = () => {
+  const [
+    oldPasswordErrors,
+    setOldPasswordErrors,
+  ] = useState([]);
+  const [
+    newPasswordErrors,
+    setNewPasswordErrors,
+  ] = useState([]);
+
+  const [
+    dataForm,
+    setDataForm,
+  ] = useState({
+    oldPassword: '',
+    newPassword: '',
+  });
+  const [
+    confirmNewPassword,
+    setConfirmNewPassword,
+  ] = useState('');
+
+  const [
+    passwordChangeSuccessModalVisible,
+    setPasswordChangeSuccessModalVisible,
+  ] = useState(false);
+  const {
+    setShowErrorModal,
+    drawErrorModal,
+  } = useErrorModal();
+
+  const oldPasswordInputHandler = (password) => {
+    setDataForm({
+      ...dataForm,
+      oldPassword: password,
+    });
+  };
+  const newPasswordInputHandler = (password) => {
+    setDataForm({
+      ...dataForm,
+      newPassword: password,
+    });
+  };
+  const confirmNewPasswordInputHandler = (password) => {
+    setConfirmNewPassword(password);
+  };
+
+  const checkPasswordValidation = (response) => {
+    const { message, statusCode } = response;
+    const oldPasswordErrs = [];
+    const newPasswordErrs = [];
+
+    if (dataForm.newPassword !== confirmNewPassword) {
```

Comment:
1. Input stare hasło wywala walidacje że hasło jest za krótkie, jedyne co ma robić to walidować po tym, że jest niezgodne z obecnym hasłem
2.  dałam dwa różne nowe hasła i przeszedł request => to czy nowe hasła się różnia trzeba walidować w onSubmit przed try. Wsadzić to w if else 
3. Errory powinny czyścić sie w try na końcu czyli po setPasswordChangeSuccessModalVisible(true)